File: CourseHomeworkEleven/app/src/main/java/com/example/coursehomeworknine/Weather.java
package com.example.coursehomeworknine;

import com.google.gson.JsonObject;
import com.google.gson.annotations.SerializedName;

import java.text.SimpleDateFormat;
import java.util.Calendar;
import java.util.List;

public class Weather{
    @SerializedName("main")
    private MainObject main;
    @SerializedName("weather")
    private List<WeatherObject> weather;
    @SerializedName("wind")
    private WindObject wind;
    @SerializedName("sys")
    private SunObject sys;

    public String getTemp() {
        return main.getTemp();
    }

    public String getWeatherConditions() {
        return weather.get(0).getDescription();
    }

    public String getWindSpeed() {
        return wind.getSpeed();
    }


    public String getHumidity() {
        return main.getHumidity();
    }


    public String getSunrise() {
        return getDate(sys.getSunrise(), "HH:mm:ss");
    }


    public String getSunset() {
        return getDate(sys.getSunset(), "HH:mm:ss");
    }

    public static String getDate(long seconds, String dateFormat)
    {
        SimpleDateFormat formatter = new SimpleDateFormat(dateFormat);
        Calendar calendar = Calendar.getInstance();
        calendar.setTimeInMillis(seconds*1000);
        return formatter.format(calendar.getTime());
    }


}

File: CourseHomeworkTen/app/src/main/java/com/example/coursehomeworkten/Weather.kt
package com.example.coursehomeworkten

import com.google.gson.annotations.SerializedName
import java.text.SimpleDateFormat
import java.util.*


class Weather {

    @SerializedName("main")
    private val main: MainObject? = null

    @SerializedName("weather")
    private val weather: List<WeatherObject>? = null

    @SerializedName("wind")
    private val wind: WindObject? = null

    @SerializedName("sys")
    private val sys: SunObject? = null

    fun getTemp(): String? {
        if (main != null) {
            return main.getTemp()
        }
        return null
    }

    fun getWeatherConditions(): String? {
        if(weather != null){
            return weather[0].getDescription()
        }
        return null
    }

    fun getWindSpeed(): String? {
        if (wind != null) {
            return wind.getSpeed()
        }
        return null
    }


    fun getHumidity(): String? {
        if (main != null) {
            return main.getHumidity()
        }
        return null
    }


    fun getSunrise(): String? {
        if (sys != null) {
            return getDate(sys.getSunrise(), "HH:mm:ss")
        }
        return null
    }


    fun getSunset(): String? {
        if (sys != null) {
            return getDate(sys.getSunset(), "HH:mm:ss")
        }
        return null
    }

    fun getDate(seconds: Long?, dateFormat: String?): String? {
        val formatter = SimpleDateFormat(dateFormat)
        val calendar: Calendar = Calendar.getInstance()
        if(seconds != null){
            calendar.setTimeInMillis(seconds * 1000)
        }
        return formatter.format(calendar.getTime())
    }
}